COMETwebapp/ViewModels/Components/Common/BaseDataItemTable/BaseDataItemTableViewModel.cs

```diff
@@ -217,6 +229,44 @@ protected virtual void RefreshAccessRight()
             this.IsLoading = false;
         }
 
+        /// <summary>
+        /// Displays a toast notification in the screen from a given result
+        /// </summary>
+        /// <param name="result">The result of an operation</param>
+        /// <exception cref="InvalidDataException">
+        /// Throws an <see cref="InvalidDataException" /> if the
+        /// <see cref="NotificationService" /> property is null
+        /// </exception>
+        /// <returns>A <see cref="Task" /></returns>
+        protected void DisplayToastNotificationFromResult(Result result)
```

Comment:
Could we move that to the NotificationService directly? seams more appropriate and it would make more sense to not have to duplicate this code each time a VM need to display a notification

Reply:
Indeed. As the INotificationService comes from AntDesign library, we can make DisplayToastNotificationFromResult an extension method of the INotificationService.

Reply:
The final implementation uses a sourcelist of results in INotificationService and displays the toasts through comet webapp component called NotificationComponent